clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -661,3 +629,61 @@ invariant_checker_t::print_results()
     std::cerr << "Trace invariant checks passed\n";
     return true;
 }
+
+std::string
+invariant_checker_t::pc_discontinuity_error_msg(void *shard_data, const memref_t &memref,
```

Comment:
The function name makes me think this takes in an enum of error types and turns it into a string, rather than identifying whether there's an error.  How about "check_for_pc_discontinuity()" or something else active.